libretroshare/src/pgp/pgphandler.cc

```diff
@@ -630,50 +630,61 @@ std::string PGPHandler::SaveCertificateToString(const RsPgpId& id,bool include_s
 	return makeRadixEncodedPGPKey(key,include_signatures) ;
 }
 
-bool PGPHandler::exportPublicKey(const RsPgpId& id,unsigned char *& mem_block,size_t& mem_size,bool armoured,bool include_signatures) const
+bool PGPHandler::exportPublicKey(
+        const RsPgpId& id,
+        rs_view_ptr<unsigned char>& mem_block, size_t& mem_size,
```

Comment:
Nope before the pointer ownership wasn't clear, the way it was passed made explicit that `exportPublicKey` could modify the pointer address but not who should free that memory, in theory that pointer could have been a pointer to some internal structure that was passed just for reading modifying but not freeing to the caller, `rs_view_ptr<T>` for the compiler is the same as `T*` as you can see where it is declared, it doesn't use any C++11 smart pointer feature, it is just a "decoration" to make it explicit to the reader that the ownership of the pointer is not transferred, just by reading the function signature without need of reading the implementation.

Reply:
Now it is clear to you. But it is not clear to anybody else including me. Using a proper comment above the method prototype would be much more useful IMHO.

Reply:
Even though there is a problem here: you cannot make theh caller of that funciton not responsible from deleting that pointer. Who will delete it otherwise? Doesn't make sense.

So mem_block should be allocated using malloc (as you do) and not with new[], but deleted by the client. 
Besides, the pointer being passed as a reference, it is clear even without comment that it is to be created in the function and therefore left to the client to delete. But anyway, a simple comment above the function prototype could recall that.

Reply:
clearly here the use of fancy decoration shadowed where the real memory management should take place, even for you. That's proof that it's not for the best here at least.